lib/dw/svelteChart.js

```diff
@@ -74,17 +74,11 @@ class Chart extends Store {
     dataset(ds) {
         // set a new dataset, or reset the old one if ds===true
         if (arguments.length) {
-            if (ds !== true) this._dataset_cache = ds;
-            const jsonData = typeof this._dataset_cache.list !== 'function';
+            if (ds === true) ds = this._dataset_cache;
```

Comment:
sorry, I think we also need to make sure the cache gets set in the first place. this should work now:
```suggestion
           if (ds !== true) this._dataset_cache = ds;
           else ds = this._dataset_cache;
```

Reply:
:+1: